### PR TITLE
detect-detection: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4019) ticket:

Describe changes:

- Update detect-detection unit tests to FAIL/PASS APIs

### CODE REVIEW

**Comment by hsadia538** [comment]
I have tried converting `DetectDetectionFilterTestSig1` , `DetectDetectionFilterTestSig2` , `DetectDetectionFilterTestSig3` to FAIL PASS APIs but let me know if Suricata verify tests are better in this case. Thanks

File: src/detect-detection-filter.c [hunk 1]
```diff
@@ -286,12 +286,13 @@ static int DetectDetectionFilterTestParse01 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("track by_dst,count 10,seconds 60");
-    if (df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 1;
-    }
+    FAIL_IF_NOT(df);
```
Comment (victorjulien) [inline src/detect-detection-filter.c:289]: FAIL_IF_NULL

File: src/detect-detection-filter.c [hunk 2]
```diff
@@ -304,12 +305,11 @@ static int DetectDetectionFilterTestParse02 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("track both,count 10,seconds 60");
-    if (df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF(df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) &&
```
Comment (victorjulien) [inline src/detect-detection-filter.c:308]: can you break these into 4 separate checks?

File: src/detect-detection-filter.c [hunk 3]
```diff
@@ -322,12 +322,13 @@ static int DetectDetectionFilterTestParse03 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("track by_dst, seconds 60, count 10");
-    if (df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 1;
-    }
+    FAIL_IF_NOT(df);
```
Comment (victorjulien) [inline src/detect-detection-filter.c:325]: FAIL_IF_NULL (please check elsewhere too)

File: src/detect-detection-filter.c [hunk 4]
```diff
@@ -341,12 +342,11 @@ static int DetectDetectionFilterTestParse04 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("count 10, track by_dst, seconds 60, count 10");
-    if (df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF(df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 60));
```
Comment (victorjulien) [inline src/detect-detection-filter.c:345]: break up check

File: src/detect-detection-filter.c [hunk 5]
```diff
@@ -611,24 +598,24 @@ static int DetectDetectionFilterTestSig3(void)
...
-    else {
-        if (alerts != 3)
-            printf("alerts: %d != 3: ", alerts);
-        if (drops != 3)
-            printf("drops: %d != 3: ", drops);
-    }
+    FAIL_IF_NOT(alerts == 3);
+    FAIL_IF_NOT(drops == 3);
+
+    if (alerts != 3)
```
Comment (victorjulien) [inline src/detect-detection-filter.c:604]: this can be removed

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline comments

File: src/detect-detection-filter.c (hunk 2 above)
Comment (hsadia538) [reply src/detect-detection-filter.c:308]: I tried breaking it and when I run the test I get a segmentation fault and the flow breaks.

**Comment by hsadia538** [comment]
Followed by #8088

File: src/detect-detection-filter.c (hunk 2 above)
Comment (hsadia538) [reply src/detect-detection-filter.c:308]: `static int DetectDetectionFilterTestParse02 (void)
{
    DetectThresholdData *df = NULL;
    df = DetectDetectionFilterParse("track both,count 10,seconds 60");

    FAIL_IF_NOT_NULL(df);
    FAIL_IF(df->track == TRACK_DST || df->track == TRACK_SRC);
    FAIL_IF(df->count == 10);
    FAIL_IF(df->seconds == 60);

    DetectDetectionFilterFree(NULL, df);

    PASS;
}`

**Pull request closed without merging**